IndexOps.getMapping raises exception if mapping contains "dynamic_templates"

Upon attempting to retrieve the mapping for an index which contains "dyanmic_templates" an exception is throw which appears to be caused by being unable to parse the response properly.


```
org.springframework.data.elasticsearch.UncategorizedElasticsearchException: {"index-name":{"mappings":{"dynamic_templates":[ ... ],"properties":{ ... }}}
```

Unfortunately I am working on proprietary software so I cannot provide the exact source/mapping but I have attempted to provide the relevant parts you asked for after applying some anonymization to them.

Spring Data ES version: **org.springframework.data:spring-data-elasticsearch:jar:4.1.1**

Mapping:
```json
{
  "mapping": {
    "dynamic_templates": [
      {
        "field1_subfield": {
          "path_match": "field1.subfield.*",
          "mapping": {
            "type": "keyword"
          }
        }
      }
    ],
    "properties": {
      "field1": {
        "properties": {
          "subfield": {
            "type": "object"
          }
        }
      }
    }
  }
}
```

Code:
```java
@Data
public class ExistsFilter implements Filter {
    ...
    ...
    ...

    @Override
    public Mono<FilterValue> getFilterValues(Query query, ReactiveElasticsearchOperations operations) {
        return operations.indexOps(Entity.class).getMapping()
                .map(mapping -> this.createFilterValue(mapping))
    }
}
```

Stack Trace
```

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v2.4.0)

2021-03-05 11:16:57.736  INFO 15008 --- [           main] c.o.p.appname.SpringApplication          : Starting SpringApplication using Java 11 on {{OMITTED}}
2021-03-05 11:16:57.743  INFO 15008 --- [           main] c.o.p.appname.SpringApplication          : The following profiles are active: dev
2021-03-05 11:16:58.359  INFO 15008 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Reactive Elasticsearch repositories in DEFAULT mode.
2021-03-05 11:16:58.512  INFO 15008 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 150 ms. Found 2 Reactive Elasticsearch repository interfaces.
2021-03-05 11:16:58.887  INFO 15008 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data Elasticsearch repositories in DEFAULT mode.
2021-03-05 11:16:58.895  INFO 15008 --- [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 7 ms. Found 0 Elasticsearch repository interfaces.
2021-03-05 11:16:59.066  INFO 15008 --- [           main] o.s.c.a.ConfigurationClassEnhancer       : @Bean method FunctionConfiguration.po is non-static and returns an object assignable to Spring's BeanFactoryPostProcessor interface. This will result in a failure to process annotations such as @Autowired, @Resource and @PostConstruct within the method's declaring @Configuration class. Add the 'static' modifier to this method to avoid these container lifecycle issues; see @Bean javadoc for complete details.
2021-03-05 11:16:59.078  INFO 15008 --- [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'errorChannel' has been explicitly defined. Therefore, a default PublishSubscribeChannel will be created.
2021-03-05 11:16:59.082  INFO 15008 --- [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'taskScheduler' has been explicitly defined. Therefore, a default ThreadPoolTaskScheduler will be created.
2021-03-05 11:16:59.086  INFO 15008 --- [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'integrationHeaderChannelRegistry' has been explicitly defined. Therefore, a default DefaultHeaderChannelRegistry will be created.
2021-03-05 11:16:59.143  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.integration.config.IntegrationManagementConfiguration' of type [org.springframework.integration.config.IntegrationManagementConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:16:59.147  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration$IntegrationJmxConfiguration' of type [org.springframework.boot.autoconfigure.integration.IntegrationAutoConfiguration$IntegrationJmxConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:16:59.156  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration' of type [org.springframework.boot.autoconfigure.jmx.JmxAutoConfiguration] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:16:59.225  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'mbeanServer' of type [com.sun.jmx.mbeanserver.JmxMBeanServer] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:16:59.243  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'integrationChannelResolver' of type [org.springframework.integration.support.channel.BeanFactoryChannelResolver] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:16:59.245  INFO 15008 --- [           main] trationDelegate$BeanPostProcessorChecker : Bean 'integrationDisposableAutoCreatedBeans' of type [org.springframework.integration.config.annotation.Disposables] is not eligible for getting processed by all BeanPostProcessors (for example: not eligible for auto-proxying)
2021-03-05 11:17:00.262  INFO 15008 --- [ctor-http-nio-2] o.s.d.elasticsearch.support.VersionInfo  : Version Spring Data Elasticsearch: 4.1.1
2021-03-05 11:17:00.264  INFO 15008 --- [ctor-http-nio-2] o.s.d.elasticsearch.support.VersionInfo  : Version Elasticsearch Client in build: 7.9.3
2021-03-05 11:17:00.264  INFO 15008 --- [ctor-http-nio-2] o.s.d.elasticsearch.support.VersionInfo  : Version Elasticsearch Client used: 7.9.3
2021-03-05 11:17:00.264  INFO 15008 --- [ctor-http-nio-2] o.s.d.elasticsearch.support.VersionInfo  : Version Elasticsearch cluster: 7.3.2
2021-03-05 11:17:00.264  WARN 15008 --- [ctor-http-nio-2] o.s.d.elasticsearch.support.VersionInfo  : Version mismatch in between Elasticsearch Client and Cluster: 7.9.3 - 7.3.2
2021-03-05 11:17:01.720  INFO 15008 --- [           main] o.s.s.c.ThreadPoolTaskScheduler          : Initializing ExecutorService 'taskScheduler'
2021-03-05 11:17:01.746  INFO 15008 --- [           main] o.s.b.a.e.web.EndpointLinksResolver      : Exposing 2 endpoint(s) beneath base path '/actuator'
false
2021-03-05 11:17:03.426  INFO 15008 --- [           main] c.o.p.appname.SpringApplication          : Started SpringApplication in 6.175 seconds (JVM running for 6.996)
2021-03-05 11:17:03.794 ERROR 15008 --- [ctor-http-nio-2] a.w.r.e.AbstractErrorWebExceptionHandler : [886daa2b-1]  500 Server Error for HTTP POST "/search"

org.springframework.data.elasticsearch.UncategorizedElasticsearchException: {{ INDEX MAPPING OUTPUT FROM ABOVE }}
	at org.springframework.data.elasticsearch.core.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:66) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 ElasticsearchExceptionTranslator.java
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Assembly trace from producer [reactor.core.publisher.FluxOnErrorResume] :
	reactor.core.publisher.Flux.onErrorMap
	org.springframework.data.elasticsearch.core.ReactiveElasticsearchTemplate.executeWithIndicesClient(ReactiveElasticsearchTemplate.java:879)
Error has been observed at the following site(s):
	|_        Flux.onErrorMap ⇢ at org.springframework.data.elasticsearch.core.ReactiveElasticsearchTemplate.executeWithIndicesClient(ReactiveElasticsearchTemplate.java:879)
 ReactiveElasticsearchTemplate.java
	|_              Mono.from ⇢ at org.springframework.data.elasticsearch.core.DefaultReactiveIndexOperations.getMapping(DefaultReactiveIndexOperations.java:181)
 DefaultReactiveIndexOperations.java
	|_           Mono.flatMap ⇢ at org.springframework.data.elasticsearch.core.DefaultReactiveIndexOperations.getMapping(DefaultReactiveIndexOperations.java:182)
 DefaultReactiveIndexOperations.java
	|_               Mono.map ⇢ at com.orgname.projname.appname.search.query.filter.ExistsFilter.getFilterValues(ExistsFilter.java:62)
 ExistsFilter.java
	|_                        ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.lambda$0(ReactiveElasticsearchSearchRepositoryImpl.java:52)
 ReactiveElasticsearchSearchRepositoryImpl.java
	|_               Mono.zip ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.lambda$1(ReactiveElasticsearchSearchRepositoryImpl.java:54)
 ReactiveElasticsearchSearchRepositoryImpl.java
	|_     Mono.switchIfEmpty ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.lambda$1(ReactiveElasticsearchSearchRepositoryImpl.java:62)
 ReactiveElasticsearchSearchRepositoryImpl.java
	|_           Mono.flatMap ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.collectFilterValues(ReactiveElasticsearchSearchRepositoryImpl.java:54)
 ReactiveElasticsearchSearchRepositoryImpl.java
	|_           Mono.zipWith ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.searchForPage(ReactiveElasticsearchSearchRepositoryImpl.java:69)
	|_               Mono.map ⇢ at com.orgname.projname.appname.elasticsearch.repository.ReactiveElasticsearchSearchRepositoryImpl.searchForPage(ReactiveElasticsearchSearchRepositoryImpl.java:70)
 ReactiveElasticsearchSearchRepositoryImpl.java
	|_                        ⇢ at org.springframework.data.repository.core.support.RepositoryMethodInvoker$ReactiveInvocationListenerDecorator.lambda$decorate$1(RepositoryMethodInvoker.java:229)
 RepositoryMethodInvoker.java
	|_         Mono.usingWhen ⇢ at org.springframework.data.repository.core.support.RepositoryMethodInvoker$ReactiveInvocationListenerDecorator.decorate(RepositoryMethodInvoker.java:226)
 RepositoryMethodInvoker.java
	|_               Mono.map ⇢ at com.orgname.projname.appname.service.SearchService.searchForPage(SearchService.java:72)
 SearchService.java
	|_                        ⇢ at com.orgname.projname.appname.controller.SearchController.search(SearchController.java:46)
 SearchController.java
	|_              Mono.from ⇢ at org.springframework.http.codec.json.AbstractJackson2Encoder.encode(AbstractJackson2Encoder.java:145)
 AbstractJackson2Encoder.java
	|_               Mono.map ⇢ at org.springframework.http.codec.json.AbstractJackson2Encoder.encode(AbstractJackson2Encoder.java:146)
 AbstractJackson2Encoder.java
	|_              Mono.flux ⇢ at org.springframework.http.codec.json.AbstractJackson2Encoder.encode(AbstractJackson2Encoder.java:147)
 AbstractJackson2Encoder.java
	|_     Flux.singleOrEmpty ⇢ at org.springframework.http.codec.EncoderHttpMessageWriter.write(EncoderHttpMessageWriter.java:122)
 EncoderHttpMessageWriter.java
	|_     Mono.switchIfEmpty ⇢ at org.springframework.http.codec.EncoderHttpMessageWriter.write(EncoderHttpMessageWriter.java:123)
 EncoderHttpMessageWriter.java
	|_           Mono.flatMap ⇢ at org.springframework.http.codec.EncoderHttpMessageWriter.write(EncoderHttpMessageWriter.java:127)
 EncoderHttpMessageWriter.java
	|_                        ⇢ at org.springframework.http.codec.EncoderHttpMessageWriter.write(EncoderHttpMessageWriter.java:203)
 EncoderHttpMessageWriter.java
	|_                        ⇢ at org.springframework.web.reactive.result.method.annotation.AbstractMessageWriterResultHandler.writeBody(AbstractMessageWriterResultHandler.java:104)
 AbstractMessageWriterResultHandler.java
	|_                        ⇢ at org.springframework.web.reactive.result.method.annotation.ResponseBodyResultHandler.handleResult(ResponseBodyResultHandler.java:86)
 ResponseBodyResultHandler.java
	|_             checkpoint ⇢ Handler com.orgname.projname.appname.controller.SearchController#search(PagedDocumentQuery) [DispatcherHandler]
	|_           Mono.flatMap ⇢ at org.springframework.web.reactive.DispatcherHandler.lambda$handleResult$5(DispatcherHandler.java:172)
 DispatcherHandler.java
	|_     Mono.onErrorResume ⇢ at org.springframework.web.reactive.DispatcherHandler.handleResult(DispatcherHandler.java:171)
 DispatcherHandler.java
	|_                        ⇢ at org.springframework.web.reactive.DispatcherHandler.lambda$handle$2(DispatcherHandler.java:147)
 DispatcherHandler.java
	|_           Mono.flatMap ⇢ at org.springframework.web.reactive.DispatcherHandler.handle(DispatcherHandler.java:147)
 DispatcherHandler.java
	|_                        ⇢ at org.springframework.web.server.handler.DefaultWebFilterChain.lambda$filter$0(DefaultWebFilterChain.java:120)
 DefaultWebFilterChain.java
	|_             Mono.defer ⇢ at org.springframework.web.server.handler.DefaultWebFilterChain.filter(DefaultWebFilterChain.java:119)
 DefaultWebFilterChain.java
	|_       Mono.doOnSuccess ⇢ at org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter.filter(MetricsWebFilter.java:78)
 MetricsWebFilter.java
	|_         Mono.doOnError ⇢ at org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter.filter(MetricsWebFilter.java:79)
 MetricsWebFilter.java
	|_ Mono.transformDeferred ⇢ at org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter.filter(MetricsWebFilter.java:73)
 MetricsWebFilter.java
	|_             checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_                        ⇢ at org.springframework.web.server.handler.DefaultWebFilterChain.lambda$filter$0(DefaultWebFilterChain.java:120)
 DefaultWebFilterChain.java
	|_             Mono.defer ⇢ at org.springframework.web.server.handler.DefaultWebFilterChain.filter(DefaultWebFilterChain.java:119)
 DefaultWebFilterChain.java
	|_                        ⇢ at org.springframework.web.server.handler.FilteringWebHandler.handle(FilteringWebHandler.java:59)
 FilteringWebHandler.java
	|_                        ⇢ at org.springframework.web.server.handler.WebHandlerDecorator.handle(WebHandlerDecorator.java:56)
	|_             Mono.error ⇢ at org.springframework.web.server.handler.ExceptionHandlingWebHandler$CheckpointInsertingHandler.handle(ExceptionHandlingWebHandler.java:98)
 ExceptionHandlingWebHandler.java
	|_             checkpoint ⇢ HTTP POST "/search" [ExceptionHandlingWebHandler]
	|_                        ⇢ at org.springframework.web.server.handler.ExceptionHandlingWebHandler.lambda$handle$0(ExceptionHandlingWebHandler.java:77)
 ExceptionHandlingWebHandler.java
	|_     Mono.onErrorResume ⇢ at org.springframework.web.server.handler.ExceptionHandlingWebHandler.handle(ExceptionHandlingWebHandler.java:77)
 ExceptionHandlingWebHandler.java
Stack trace:
		at org.springframework.data.elasticsearch.core.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:66) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 ElasticsearchExceptionTranslator.java
		at org.springframework.data.elasticsearch.core.ReactiveElasticsearchTemplate.translateException(ReactiveElasticsearchTemplate.java:946) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 ReactiveElasticsearchTemplate.java
		at reactor.core.publisher.Flux.lambda$onErrorMap$27(Flux.java:6544) ~[reactor-core-3.4.0.jar:3.4.0]
 Flux.java
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxOnErrorResume.java
		at reactor.core.publisher.MonoNext$NextSubscriber.onError(MonoNext.java:93) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoNext.java
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxOnErrorResume.java
		at reactor.core.publisher.Operators.error(Operators.java:196) ~[reactor-core-3.4.0.jar:3.4.0]
 Operators.java
		at reactor.core.publisher.MonoError.subscribe(MonoError.java:52) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoError.java
		at reactor.core.publisher.Mono.subscribe(Mono.java:3987) ~[reactor-core-3.4.0.jar:3.4.0]
 Mono.java
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxOnErrorResume.java
		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:106) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxOnErrorResume.java
		at reactor.core.publisher.MonoIgnoreThen$ThenIgnoreMain.drain(MonoIgnoreThen.java:139) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoIgnoreThen.java
		at reactor.core.publisher.MonoIgnoreThen.subscribe(MonoIgnoreThen.java:56) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoIgnoreThen.java
		at reactor.core.publisher.Mono.subscribe(Mono.java:3987) ~[reactor-core-3.4.0.jar:3.4.0]
 Mono.java
		at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:103) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxOnErrorResume.java
		at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onError(Operators.java:2023) ~[reactor-core-3.4.0.jar:3.4.0]
 Operators.java
		at reactor.core.publisher.MonoFlatMap$FlatMapMain.onError(MonoFlatMap.java:172) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:142) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:210) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxPeekFuseable.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:127) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxContextWrite.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:295) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:337) ~[reactor-core-3.4.0.jar:3.4.0]
		at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1784) ~[reactor-core-3.4.0.jar:3.4.0]
 Operators.java
		at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:159) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoCollect.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onComplete(FluxPeekFuseable.java:277) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxPeekFuseable.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:383) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 FluxReceive.java
		at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:396) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperations.java
		at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:452) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperations.java
		at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:664) ~[reactor-netty-http-1.0.1.jar:1.0.1]
 HttpClientOperations.java
		at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:94) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperationsHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[netty-handler-4.1.54.Final.jar:4.1.54.Final]
 IdleStateHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 MessageToMessageDecoder.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 CombinedChannelDuplexHandler.java
		at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 ByteToMessageDecoder.java
		at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 ByteToMessageDecoder.java
		at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 CombinedChannelDuplexHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
		at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 DefaultChannelPipeline.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 DefaultChannelPipeline.java
		at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractNioByteChannel.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 SingleThreadEventExecutor.java
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 ThreadExecutorMap.java
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 FastThreadLocalRunnable.java
		at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
 Thread.java
Caused by: org.elasticsearch.ElasticsearchStatusException: {"search-api-document":{"mappings":{"dynamic_templates":[{"ner_labels":{"path_match":"ner.labels.*","mapping":{"type":"keyword"}}}],"properties":{"catalogId":{"type":"keyword"},"catalogVersion":{"type":"keyword"},"corpus":{"type":"keyword"},"createdAt":{"type":"date","format":"basic_date_time"},"keyphrases":{"properties":{"count":{"type":"long"},"phrase":{"type":"text"},"score":{"type":"double"}}},"metadata":{"type":"object"},"ner":{"properties":{"availableLabels":{"type":"keyword"},"labels":{"type":"object"},"mergedLabelsJson":{"type":"text","index":false},"modelId":{"type":"keyword"},"modelVersion":{"type":"keyword"},"oovCount":{"type":"long"},"state":{"type":"keyword"}}},"sourceLink":{"type":"keyword"},"summary":{"type":"text"},"text":{"type":"text","store":true,"analyzer":"standard_synonym"},"updatedAt":{"type":"date","format":"basic_date_time"},"versionNo":{"type":"keyword"}}}}}
	at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.doDecode(DefaultReactiveElasticsearchClient.java:793) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 DefaultReactiveElasticsearchClient.java
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Assembly trace from producer [reactor.core.publisher.MonoFlatMap] :
	reactor.core.publisher.Mono.flatMap
	org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.readResponseBody(DefaultReactiveElasticsearchClient.java:768)
Error has been observed at the following site(s):
	|_       Mono.flatMap ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.readResponseBody(DefaultReactiveElasticsearchClient.java:768)
 DefaultReactiveElasticsearchClient.java
	|_          Mono.from ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.lambda$null$18(DefaultReactiveElasticsearchClient.java:578)
 DefaultReactiveElasticsearchClient.java
	|_       Mono.flatMap ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient$DefaultRequestBodyUriSpec.lambda$exchangeToMono$3(DefaultWebClient.java:397)
 DefaultWebClient.java
	|_ Mono.switchIfEmpty ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient$DefaultRequestBodyUriSpec.lambda$exchangeToMono$3(DefaultWebClient.java:398)
	|_          Mono.then ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient.releaseIfNotConsumed(DefaultWebClient.java:155)
 DefaultWebClient.java
	|_                    ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient.access$1000(DefaultWebClient.java:64)
 DefaultWebClient.java
	|_                    ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient$DefaultRequestBodyUriSpec.lambda$null$2(DefaultWebClient.java:399)
 DefaultWebClient.java
	|_ Mono.onErrorResume ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient$DefaultRequestBodyUriSpec.lambda$exchangeToMono$3(DefaultWebClient.java:399)
 DefaultWebClient.java
	|_       Mono.flatMap ⇢ at org.springframework.web.reactive.function.client.DefaultWebClient$DefaultRequestBodyUriSpec.exchangeToMono(DefaultWebClient.java:394)
 DefaultWebClient.java
	|_                    ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.lambda$sendRequest$19(DefaultReactiveElasticsearchClient.java:576)
 DefaultReactiveElasticsearchClient.java
	|_       Mono.flatMap ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.execute(DefaultReactiveElasticsearchClient.java:535)
 DefaultReactiveElasticsearchClient.java
	|_         Mono.error ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.lambda$execute$16(DefaultReactiveElasticsearchClient.java:544)
 DefaultReactiveElasticsearchClient.java
	|_ Mono.onErrorResume ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.execute(DefaultReactiveElasticsearchClient.java:536)
 DefaultReactiveElasticsearchClient.java
	|_          Flux.from ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.sendRequest(DefaultReactiveElasticsearchClient.java:576)
 DefaultReactiveElasticsearchClient.java
	|_                    ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.sendRequest(DefaultReactiveElasticsearchClient.java:568)
 DefaultReactiveElasticsearchClient.java
	|_          Flux.next ⇢ at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.getMapping(DefaultReactiveElasticsearchClient.java:674)
 DefaultReactiveElasticsearchClient.java
	|_                    ⇢ at org.springframework.data.elasticsearch.client.reactive.ReactiveElasticsearchClient$Indices.getMapping(ReactiveElasticsearchClient.java:1149)
 ReactiveElasticsearchClient.java
	|_         Flux.defer ⇢ at org.springframework.data.elasticsearch.core.ReactiveElasticsearchTemplate.executeWithIndicesClient(ReactiveElasticsearchTemplate.java:879)
 ReactiveElasticsearchTemplate.java
Stack trace:
		at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.doDecode(DefaultReactiveElasticsearchClient.java:793) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 DefaultReactiveElasticsearchClient.java
		at org.springframework.data.elasticsearch.client.reactive.DefaultReactiveElasticsearchClient.lambda$readResponseBody$27(DefaultReactiveElasticsearchClient.java:768) ~[spring-data-elasticsearch-4.1.1.jar:4.1.1]
 DefaultReactiveElasticsearchClient.java
		at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoFlatMap.java
		at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onNext(FluxPeekFuseable.java:210) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxPeekFuseable.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:127) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxContextWrite.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableConditionalSubscriber.onNext(FluxMapFuseable.java:295) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxFilterFuseable$FilterFuseableConditionalSubscriber.onNext(FluxFilterFuseable.java:337) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxFilterFuseable.java
		at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1784) ~[reactor-core-3.4.0.jar:3.4.0]
 Operators.java
		at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:159) ~[reactor-core-3.4.0.jar:3.4.0]
 MonoCollect.java
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.core.publisher.FluxPeekFuseable$PeekFuseableSubscriber.onComplete(FluxPeekFuseable.java:277) ~[reactor-core-3.4.0.jar:3.4.0]
		at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150) ~[reactor-core-3.4.0.jar:3.4.0]
 FluxMapFuseable.java
		at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:383) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 FluxReceive.java
		at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:396) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperations.java
		at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:452) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperations.java
		at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:664) ~[reactor-netty-http-1.0.1.jar:1.0.1]
 HttpClientOperations.java
		at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:94) ~[reactor-netty-core-1.0.1.jar:1.0.1]
 ChannelOperationsHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) ~[netty-handler-4.1.54.Final.jar:4.1.54.Final]
 IdleStateHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 MessageToMessageDecoder.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 CombinedChannelDuplexHandler.java
		at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 ByteToMessageDecoder.java
		at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-4.1.54.Final.jar:4.1.54.Final]
 ByteToMessageDecoder.java
		at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 CombinedChannelDuplexHandler.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 DefaultChannelPipeline.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractChannelHandlerContext.java
		at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
		at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 DefaultChannelPipeline.java
		at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 AbstractNioByteChannel.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493) ~[netty-transport-4.1.54.Final.jar:4.1.54.Final]
 NioEventLoop.java
		at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 SingleThreadEventExecutor.java
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 ThreadExecutorMap.java
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.54.Final.jar:4.1.54.Final]
 FastThreadLocalRunnable.java
		at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
 Thread.java
```

--- a/src/main/java/org/springframework/data/elasticsearch/client/reactive/DefaultReactiveElasticsearchClient.java
+++ b/src/main/java/org/springframework/data/elasticsearch/client/reactive/DefaultReactiveElasticsearchClient.java
@@ -685,6 +685,7 @@ public class DefaultReactiveElasticsearchClient implements ReactiveElasticsearch
 	}
 
 	@Override
+	@Deprecated
 	public Mono<org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse> getMapping(HttpHeaders headers,
 			org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest getMappingsRequest) {
 		return sendRequest(getMappingsRequest, requestCreator.getMapping(),

--- a/src/main/java/org/springframework/data/elasticsearch/client/util/RequestConverters.java
+++ b/src/main/java/org/springframework/data/elasticsearch/client/util/RequestConverters.java
@@ -812,6 +812,7 @@ public class RequestConverters {
 		return request;
 	}
 
+	@Deprecated
 	public static Request putMapping(PutMappingRequest putMappingRequest) {
 		// The concreteIndex is an internal concept, not applicable to requests made over the REST API.
 		if (putMappingRequest.getConcreteIndex() != null) {

--- a/src/main/java/org/springframework/data/elasticsearch/core/AbstractDefaultIndexOperations.java
+++ b/src/main/java/org/springframework/data/elasticsearch/core/AbstractDefaultIndexOperations.java
@@ -184,10 +184,12 @@ abstract class AbstractDefaultIndexOperations implements IndexOperations {
 	protected abstract void doRefresh(IndexCoordinates indexCoordinates);
 
 	@Override
+	@Deprecated
 	public boolean addAlias(AliasQuery query) {
 		return doAddAlias(query, getIndexCoordinates());
 	}
 
+	@Deprecated
 	protected abstract boolean doAddAlias(AliasQuery query, IndexCoordinates index);
 
 	@Override
@@ -198,10 +200,12 @@ abstract class AbstractDefaultIndexOperations implements IndexOperations {
 	protected abstract List<AliasMetadata> doQueryForAlias(IndexCoordinates index);
 
 	@Override
+	@Deprecated
 	public boolean removeAlias(AliasQuery query) {
 		return doRemoveAlias(query, getIndexCoordinates());
 	}
 
+	@Deprecated
 	protected abstract boolean doRemoveAlias(AliasQuery query, IndexCoordinates index);
 
 	@Override

--- a/src/main/java/org/springframework/data/elasticsearch/core/DefaultIndexOperations.java
+++ b/src/main/java/org/springframework/data/elasticsearch/core/DefaultIndexOperations.java
@@ -141,6 +141,7 @@ class DefaultIndexOperations extends AbstractDefaultIndexOperations implements I
 	}
 
 	@Override
+	@Deprecated
 	protected boolean doAddAlias(AliasQuery query, IndexCoordinates index) {
 
 		IndicesAliasesRequest request = requestFactory.indicesAddAliasesRequest(query, index);
@@ -149,6 +150,7 @@ class DefaultIndexOperations extends AbstractDefaultIndexOperations implements I
 	}
 
 	@Override
+	@Deprecated
 	protected boolean doRemoveAlias(AliasQuery query, IndexCoordinates index) {
 
 		Assert.notNull(index, "No index defined for Alias");

--- a/src/main/java/org/springframework/data/elasticsearch/core/DefaultReactiveIndexOperations.java
+++ b/src/main/java/org/springframework/data/elasticsearch/core/DefaultReactiveIndexOperations.java
@@ -27,13 +27,13 @@ import java.util.Set;
 import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest;
 import org.elasticsearch.action.admin.indices.alias.get.GetAliasesRequest;
 import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
-import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
-import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest;
 import org.elasticsearch.action.admin.indices.settings.get.GetSettingsRequest;
 import org.elasticsearch.action.admin.indices.template.delete.DeleteIndexTemplateRequest;
 import org.elasticsearch.client.GetAliasesResponse;
 import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.GetIndexRequest;
 import org.elasticsearch.client.indices.GetIndexTemplatesRequest;
+import org.elasticsearch.client.indices.GetMappingsRequest;
 import org.elasticsearch.client.indices.IndexTemplatesExistRequest;
 import org.elasticsearch.client.indices.PutIndexTemplateRequest;
 import org.slf4j.Logger;
@@ -153,7 +153,7 @@ class DefaultReactiveIndexOperations implements ReactiveIndexOperations {
 	@Override
 	public Mono<Boolean> exists() {
 
-		GetIndexRequest request = requestFactory.getIndexRequestReactive(getIndexCoordinates().getIndexName());
+		GetIndexRequest request = requestFactory.getIndexRequest(getIndexCoordinates());
 		return Mono.from(operations.executeWithIndicesClient(client -> client.existsIndex(request)));
 	}
 
@@ -185,7 +185,7 @@ class DefaultReactiveIndexOperations implements ReactiveIndexOperations {
 
 	@Override
 	public Mono<Boolean> putMapping(Mono<Document> mapping) {
-		return mapping.map(document -> requestFactory.putMappingRequestReactive(getIndexCoordinates(), document)) //
+		return mapping.map(document -> requestFactory.putMappingRequest(getIndexCoordinates(), document)) //
 				.flatMap(request -> Mono.from(operations.executeWithIndicesClient(client -> client.putMapping(request))));
 	}
 
@@ -193,13 +193,13 @@ class DefaultReactiveIndexOperations implements ReactiveIndexOperations {
 	public Mono<Document> getMapping() {
 
 		IndexCoordinates indexCoordinates = getIndexCoordinates();
-		GetMappingsRequest request = requestFactory.getMappingRequestReactive(indexCoordinates);
+		GetMappingsRequest request = requestFactory.getMappingsRequest(indexCoordinates);
 
 		return Mono.from(operations.executeWithIndicesClient(client -> client.getMapping(request)))
 				.flatMap(getMappingsResponse -> {
-					Document document = Document.create();
-					document.put("properties",
-							getMappingsResponse.mappings().get(indexCoordinates.getIndexName()).get("properties").getSourceAsMap());
+					Map<String, Object> source = getMappingsResponse.mappings().get(indexCoordinates.getIndexName())
+							.getSourceAsMap();
+					Document document = Document.from(source);
 					return Mono.just(document);
 				});
 	}

--- a/src/main/java/org/springframework/data/elasticsearch/core/DefaultTransportIndexOperations.java
+++ b/src/main/java/org/springframework/data/elasticsearch/core/DefaultTransportIndexOperations.java
@@ -153,6 +153,7 @@ class DefaultTransportIndexOperations extends AbstractDefaultIndexOperations imp
 	}
 
 	@Override
+	@Deprecated
 	protected boolean doRemoveAlias(AliasQuery query, IndexCoordinates index) {
 
 		Assert.notNull(index, "No index defined for Alias");

--- a/src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java
+++ b/src/main/java/org/springframework/data/elasticsearch/core/RequestFactory.java
@@ -133,6 +133,7 @@ class RequestFactory {
 	}
 
 	// region alias
+	@Deprecated
 	public IndicesAliasesRequest.AliasActions aliasAction(AliasQuery query, IndexCoordinates index) {
 
 		Assert.notNull(index, "No index defined for Alias");
@@ -178,6 +179,7 @@ class RequestFactory {
 		return getAliasesRequest;
 	}
 
+	@Deprecated
 	public IndicesAliasesRequest indicesAddAliasesRequest(AliasQuery query, IndexCoordinates index) {
 		IndicesAliasesRequest.AliasActions aliasAction = aliasAction(query, index);
 		IndicesAliasesRequest request = new IndicesAliasesRequest();
@@ -253,6 +255,7 @@ class RequestFactory {
 		return requestBuilder;
 	}
 
+	@Deprecated
 	public IndicesAliasesRequest indicesRemoveAliasesRequest(AliasQuery query, IndexCoordinates index) {
 
 		String[] indexNames = index.getIndexNames();
@@ -264,6 +267,7 @@ class RequestFactory {
 				.addAliasAction(aliasAction);
 	}
 
+	@Deprecated
 	IndicesAliasesRequestBuilder indicesRemoveAliasesRequestBuilder(Client client, AliasQuery query,
 			IndexCoordinates index) {
 
@@ -347,26 +351,7 @@ class RequestFactory {
 	// endregion
 
 	// region index management
-	/**
-	 * creates a CreateIndexRequest from the rest-high-level-client library.
-	 *
-	 * @param index name of the index
-	 * @param settings optional settings
-	 * @return request
-	 */
-	public CreateIndexRequest createIndexRequest(IndexCoordinates index, @Nullable Document settings) {
-		return createIndexRequest(index, settings, null);
-	}
 
-	/**
-	 * creates a CreateIndexRequest from the rest-high-level-client library.
-	 *
-	 * @param index name of the index
-	 * @param settings optional settings
-	 * @param mapping optional mapping
-	 * @return request
-	 * @since 4.2
-	 */
 	public CreateIndexRequest createIndexRequest(IndexCoordinates index, @Nullable Document settings, @Nullable Document mapping) {
 		CreateIndexRequest request = new CreateIndexRequest(index.getIndexName());
 
@@ -379,38 +364,6 @@ class RequestFactory {
 		}
 
 		return request;
-	}
-
-	/**
-	 * creates a CreateIndexRequest from the elasticsearch library, used by the reactive methods.
-	 *
-	 * @param indexName name of the index
-	 * @param settings optional settings
-	 * @return request
-	 */
-	public org.elasticsearch.action.admin.indices.create.CreateIndexRequest createIndexRequestReactive(String indexName,
-			@Nullable Document settings) {
-
-		org.elasticsearch.action.admin.indices.create.CreateIndexRequest request = new org.elasticsearch.action.admin.indices.create.CreateIndexRequest(
-				indexName);
-		request.index(indexName);
-
-		if (settings != null && !settings.isEmpty()) {
-			request.settings(settings);
-		}
-		return request;
-	}
-
-	public CreateIndexRequestBuilder createIndexRequestBuilder(Client client, IndexCoordinates index,
-			@Nullable Document settings) {
-
-		String indexName = index.getIndexName();
-		CreateIndexRequestBuilder createIndexRequestBuilder = client.admin().indices().prepareCreate(indexName);
-
-		if (settings != null) {
-			createIndexRequestBuilder.setSettings(settings);
-		}
-		return createIndexRequestBuilder;
 	}
 
 	public CreateIndexRequestBuilder createIndexRequestBuilder(Client client, IndexCoordinates index,
@@ -431,27 +384,8 @@ class RequestFactory {
 		return createIndexRequestBuilder;
 	}
 
-	/**
-	 * creates a GetIndexRequest from the rest-high-level-client library.
-	 *
-	 * @param index name of the index
-	 * @return request
-	 */
 	public GetIndexRequest getIndexRequest(IndexCoordinates index) {
 		return new GetIndexRequest(index.getIndexNames());
-	}
-
-	/**
-	 * creates a CreateIndexRequest from the elasticsearch library, used by the reactive methods.
-	 *
-	 * @param indexName name of the index
-	 * @return request
-	 */
-	public org.elasticsearch.action.admin.indices.get.GetIndexRequest getIndexRequestReactive(String indexName) {
-
-		org.elasticsearch.action.admin.indices.get.GetIndexRequest request = new org.elasticsearch.action.admin.indices.get.GetIndexRequest();
-		request.indices(indexName);
-		return request;
 	}
 
 	public IndicesExistsRequest indicesExistsRequest(IndexCoordinates index) {
@@ -485,15 +419,6 @@ class RequestFactory {
 		return request;
 	}
 
-	public org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest putMappingRequestReactive(
-			IndexCoordinates index, Document mapping) {
-		org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest request = new org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest(
-				index.getIndexName());
-		request.type("not-used-but-must-be-there");
-		request.source(mapping);
-		return request;
-	}
-
 	public PutMappingRequestBuilder putMappingRequestBuilder(Client client, IndexCoordinates index, Document mapping) {
 
 		String[] indexNames = index.getIndexNames();
@@ -501,13 +426,6 @@ class RequestFactory {
 				.setType(IndexCoordinates.TYPE);
 		requestBuilder.setSource(mapping);
 		return requestBuilder;
-	}
-
-	public org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest getMappingRequestReactive(
-			IndexCoordinates index) {
-		org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest request = new org.elasticsearch.action.admin.indices.mapping.get.GetMappingsRequest();
-		request.indices(index.getIndexName());
-		return request;
 	}
 
 	public GetSettingsRequest getSettingsRequest(String indexName, boolean includeDefaults) {

--- a/src/test/java/org/springframework/data/elasticsearch/core/ReactiveIndexOperationsTest.java
+++ b/src/test/java/org/springframework/data/elasticsearch/core/ReactiveIndexOperationsTest.java
@@ -51,7 +51,6 @@ import org.springframework.data.elasticsearch.core.index.GetTemplateRequest;
 import org.springframework.data.elasticsearch.core.index.PutTemplateRequest;
 import org.springframework.data.elasticsearch.core.index.TemplateData;
 import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates;
-import org.springframework.data.elasticsearch.junit.jupiter.ElasticsearchRestTemplateConfiguration;
 import org.springframework.data.elasticsearch.junit.jupiter.ReactiveElasticsearchRestTemplateConfiguration;
 import org.springframework.data.elasticsearch.junit.jupiter.SpringIntegrationTest;
 import org.springframework.lang.Nullable;
@@ -67,7 +66,7 @@ public class ReactiveIndexOperationsTest {
 	public static final String TESTINDEX = "reactive-index-operations-testindex";
 
 	@Configuration
-	@Import({ ReactiveElasticsearchRestTemplateConfiguration.class, ElasticsearchRestTemplateConfiguration.class })
+	@Import({ ReactiveElasticsearchRestTemplateConfiguration.class })
 	static class Config {}
 
 	@Autowired private ReactiveElasticsearchOperations operations;

